wagtail/contrib/search_promotions/views.py

```diff
@@ -21,9 +23,14 @@
 def index(request):
     is_searching = False
     query_string = request.GET.get('q', "")
-
```

Comment:
These changes did not work for me initially, not sure why but I have made some changes locally that appear to work, see below.

I hope this helps, there are a lot of ways to do this but I borrowed from some other ordering code in other Wagtail views.

```python
def index(request):
    is_searching = False
    query_string = request.GET.get('q', '') # unrelated - just standardising on single quotes

    # Ordering
    valid_ordering = ['query_string', '-query_string', 'ohits', '-ohits'] # set up 'valid' values early to avoid errors
    ordering = valid_ordering[0] # set the 'default' value in one place

    if 'ordering' in request.GET and request.GET['ordering'] in valid_ordering:
        ordering = request.GET['ordering'] # here is the only scenario where ordering would be used from url params

    # Query
    queries = Query.objects.filter(editors_picks__isnull=False).distinct() # set up 'base' query in one place

    if 'ohits' in ordering: # standardise on 'ohits' in both query and the annotation for clarity
        queries = queries.annotate(ohits=Sum('daily_hits__hits')) # only here do we do the annotation

    queries = queries.order_by(ordering) # finally we do the actual 'ordering' based on the above
    # ... rest below (from Search comment)
```